File: jobs/2026-07-03__03-23-57/encore_migrate_gin_go__T8smG9d/artifacts/user/myproject/hello/hello.go
package hello

import (
	"context"
)

type Response struct {
	Message string `json:"message"`
}

// Hello returns a greeting message.
//encore:api public method=GET path=/hello/:name
func Hello(ctx context.Context, name string) (*Response, error) {
	return &Response{Message: "Hello, " + name + "!"}, nil
}

// Ping returns pong.
//encore:api public method=GET path=/ping
func Ping(ctx context.Context) (*Response, error) {
	return &Response{Message: "pong"}, nil
}

File: jobs/2026-07-03__12-25-47/encore_create_hello_world_ts__WMYwevb/artifacts/user/myproject/hello.ts
import { api } from "encore.dev/api";

interface HelloResponse {
  message: string;
}

export const get = api<{ name: string }, HelloResponse>(
  { expose: true, method: "GET", path: "/hello/:name" },
  async ({ name }) => {
    return { message: `Hello ${name}!` };
  }
);
